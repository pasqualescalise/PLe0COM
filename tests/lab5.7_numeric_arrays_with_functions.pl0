VAR i, array[10] : int;
VAR byte_array[10] : byte;
VAR ubyte_array[10] : ubyte;

PROCEDURE for_in_print(arr[10], len : int);
	VAR i : int;

	BEGIN
		for i := 0; i < len; i := i + 1 do begin
			print arr[i]
		end
	END;

PROCEDURE for_in_square_and_print(arr[10], len : int);
	VAR elem, i : int;
	BEGIN
		for i := 0; i < len; i := i + 1 do begin
			elem := arr[i];
			arr[i] := elem * elem
		end;

		CALL for_in_print(arr, len)
	END;

PROCEDURE for_in_print_byte(arr[20] : byte, len : int);
	VAR i : int;

	BEGIN
		for i := 0; i < len; i := i + 1 do begin
			print arr[i]
		end
	END;

PROCEDURE for_in_cube_and_print_byte(arr[10] : byte, len : int);
	VAR elem, i : int;
	BEGIN
		for i := 0; i < len; i := i + 1 do begin
			elem := arr[i];
			arr[i] := elem * elem * elem
		end;

		CALL for_in_print_byte(arr, len)
	END;

PROCEDURE for_in_print_ubyte(arr[20] : ubyte, len : int);
	VAR i : int;

	BEGIN
		for i := 0; i < len; i := i + 1 do begin
			print arr[i]
		end
	END;

PROCEDURE for_in_cube_and_print_ubyte(arr[10] : ubyte, len : int);
	VAR elem, i : int;
	BEGIN
		for i := 0; i < len; i := i + 1 do begin
			elem := arr[i];
			arr[i] := elem * elem * elem
		end;

		CALL for_in_print_ubyte(arr, len)
	END;

BEGIN
	for i := 0; i < 10; i := i + 1 do begin
		array[i] := i
	end;
	CALL for_in_print(array, 10);
	CALL for_in_square_and_print(array, 10);
	CALL for_in_print(array, 10);

	for i := 0; i < 10; i := i + 1 do begin
		byte_array[i] := i
	end;
	CALL for_in_print_byte(byte_array, 10);
	CALL for_in_cube_and_print_byte(byte_array, 10);
	CALL for_in_print_byte(byte_array, 10);

	for i := 0; i < 10; i := i + 1 do begin
		ubyte_array[i] := i
	end;
	CALL for_in_print_ubyte(ubyte_array, 10);
	CALL for_in_cube_and_print_ubyte(ubyte_array, 10);
	CALL for_in_print_ubyte(ubyte_array, 10)
END.
