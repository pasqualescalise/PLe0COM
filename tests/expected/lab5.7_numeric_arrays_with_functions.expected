0
1
2
3
4
5
6
7
8
9
0
1
4
9
16
25
36
49
64
81
0
1
4
9
16
25
36
49
64
81
0
1
2
3
4
5
6
7
8
9
0
1
8
27
64
125
-40
87
0
-39
0
1
8
27
64
125
-40
87
0
-39
0
1
2
3
4
5
6
7
8
9
0
1
8
27
64
125
216
87
0
217
0
1
8
27
64
125
216
87
0
217
